File: 3-leetcode/02-medium/js/2275-largest-combination-with-bitwise-and-greater-than-zero.ts
const largestCombination = (candidates: number[]): number => {
  let maxCount = 0;

  for (let i = 0; i < 24; i++) {
    let count = 0;

    for (let candidate of candidates) {
      if ((candidate & (1 << i)) !== 0) {
        count++;
      }
    }

    maxCount = Math.max(maxCount, count);
  }

  return maxCount;
};

File: 3-leetcode/02-medium/dart/2275-largest-combination-with-bitwise-and-greater-than-zero.dart
class Solution {
  int largestCombination(List<int> candidates) {
    int maxCount = 0;

    for (int i = 0; i < 24; i++) {
      int count = 0;

      for (int candidate in candidates) {
        if (candidate & (1 << i) != 0) {
          count++;
        }
      }

      maxCount = max(maxCount, count);
    }

    return maxCount;
  }
}
